### ISSUE TITLE
Implement a JSON DataLoader

### ISSUE THREAD

**Issue by tr0llocaust** [zzzprojects/EntityFramework-Effort#72]
It would be really great to just pull down data from a Web API and seed a context with the saved .json response for unit testing.

**Comment by uesleilima** [comment]
Just use ObjectDataLoader from Effort.Extra and create this extension method:

```
        public static ObjectDataTable<T> Table<T>(this ObjectData data, string jsonArrayFilePath, string tableName = null)
        {
            var dataTable = data.Table<T>(tableName);

            string json = File.ReadAllText(jsonArrayFilePath);
            var dataObjects = JsonConvert.DeserializeObject<IEnumerable<T>>(json);
            foreach (T item in dataObjects)
            {
                dataTable.Add(item);
            }

            return dataTable;
        }
```

After that create DbConnection:
```
ObjectData data = new ObjectData();
data.Table<Product>(@"path\to\Products.json");
DbConnectionFactory.CreateTransient(new ObjectDataLoader(data));
```

**Comment by JonathanMagnan** [comment]
Hello @tr0llocaust ,

We will close this request since it looks to works with `Extra Effort`.

We are currently looking if we should take some code from extra to get it back in `Effort` instead. We should contact the owner soon about it.

Let me know if everything was working when you made your test with `Extra Effort`.

Best Regards,

Jonathan

---

**Performance Libraries**
`context.BulkInsert(list, options => options.BatchSize = 1000);  `
[Entity Framework Extensions](http://entityframework-extensions.net/) • [Entity Framework Classic](http://entityframework-classic.net/) • [Bulk Operations](http://bulk-operations.net/) •  [Dapper Plus](http://dapper-plus.net/) •  [LinqToSql Plus](http://linqtosql-plus.net/)

**Runtime Evaluation**
`Eval.Execute("x + y", new {x = 1, y = 2}); // return 3`
[C# Eval Function](http://eval-expression.net/) • [SQL Eval Function](http://eval-sql.net/)
